src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1546,6 +1595,81 @@ protected override void HandleThreadsRequestAsync(IRequestResponder<ThreadsArgum
             responder.SetResponse(response);
         }
 
+        public static ProtocolMessages.Module ConvertToModule(in MODULE_INFO debugModuleInfo, IDebugModule2 module, bool isLoad)
+        {
+            var path = debugModuleInfo.m_bstrUrl;
+            var vsTimestampUTC = (debugModuleInfo.dwValidFields & enum_MODULE_INFO_FIELDS.MIF_TIMESTAMP) != 0 ? FileTimeToPosix(debugModuleInfo.m_TimeStamp).ToString(CultureInfo.InvariantCulture) : null;
+            var version = debugModuleInfo.m_bstrVersion;
+            var vsLoadAddress = debugModuleInfo.m_addrLoadAddress.ToString(CultureInfo.InvariantCulture);
+            var vsPreferredLoadAddress = debugModuleInfo.m_addrPreferredLoadAddress.ToString(CultureInfo.InvariantCulture);
+            var vsModuleSize = (int)debugModuleInfo.m_dwSize;
+            var vsLoadOrder = (int)debugModuleInfo.m_dwLoadOrder;
+            var symbolFilePath = debugModuleInfo.m_bstrUrlSymbolLocation;
+            var symbolStatus = debugModuleInfo.m_bstrDebugMessage;
+            var vsIs64Bit = (debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_64BIT) != 0;
+
+            ProtocolMessages.Module mod = null;
+            if (isLoad)
+            {
+                mod = new ProtocolMessages.Module(RegisterDebugModule(module), debugModuleInfo.m_bstrName)
+                {
+                    Path = path, VsTimestampUTC = vsTimestampUTC, Version = version, VsLoadAddress = vsLoadAddress, VsPreferredLoadAddress = vsPreferredLoadAddress,
+                    VsModuleSize = vsModuleSize, VsLoadOrder = vsLoadOrder, SymbolFilePath = symbolFilePath, SymbolStatus = symbolStatus, VsIs64Bit = vsIs64Bit
+                };
+            } else {
+                mod = new ProtocolMessages.Module(module.GetHashCode(), debugModuleInfo.m_bstrName)
```

Comment:
>module.GetHashCode() [](start = 50, length = 20)

No, you need the id from the handle collection.

Reply:
As you mentioned below, using HandleCollection only allows us to retrieve an IDebugModule2 given an int but not the other way around.